speechbrain/lobes/models/discrete/SQ-Codec.py

```diff
@@ -0,0 +1,1497 @@
+
+        Returns
+        -------
+        torch.Tensor
+            The gradient of the input with respect to the loss.
+        """
+        grad_input = grad_output.clone()
+        return grad_input
+
+
+class round_func_binary(InplaceFunction):
+    """
+    A custom rounding function that rounds input values to the nearest integer.
+
+    """
+
+    @staticmethod
+    def forward(ctx, input):
+        """
+        Rounds the input tensor to the nearest integer.
+
+        Parameters
+        ----------
+        ctx : context object
+            Context object used to store information for the backward computation.
+        input : torch.Tensor
+            The input tensor to be rounded.
+
+        Returns
+        -------
+        torch.Tensor
+            The input tensor rounded to the nearest integer.
+        """
+        ctx.input = input
+        return torch.round(input)
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        """
+        Computes the gradient of the input for backpropagation.
+
+        Parameters
+        ----------
+        ctx : context object
+            Context object containing information saved during the forward pass.
+        grad_output : torch.Tensor
+            The gradient of the output with respect to the loss.
+
+        Returns
+        -------
+        torch.Tensor
+            The gradient of the input with respect to the loss.
+        """
+        grad_input = grad_output.clone()
+        return grad_input
+
+
+class Heaviside(InplaceFunction):
+    """
+    A custom function that applies the Heaviside step function to the input tensor.
+
+    """
+
+    @staticmethod
+    def forward(ctx, input):
+        """
+        Applies the Heaviside step function to the input tensor.
+
+        Parameters
+        ----------
+        ctx : context object
+            Context object used to store information for the backward computation.
+        input : torch.Tensor
+            The input tensor to which the Heaviside step function is applied.
+
+        Returns
+        -------
+        torch.Tensor
+            The input tensor with the Heaviside step function applied.
+        """
+        ctx.input = input
+        values = torch.tensor([0.0]).type_as(input)
+        return torch.heaviside(input, values)
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        """
+        Computes the gradient of the input for backpropagation.
+
+        Parameters
+        ----------
+        ctx : context object
+            Context object containing information saved during the forward pass.
+        grad_output : torch.Tensor
+            The gradient of the output with respect to the loss.
+
+        Returns
+        -------
+        torch.Tensor
+            The gradient of the input with respect to the loss.
+        """
+        grad_input = grad_output.clone()
+        return grad_input
+
+
+class PreProcessor(nn.Module):
+    """
```

Comment:
Maybe the docstring should integrate some contextualisation information about where this is used in the model.

Reply:
added